### DISCUSSION TITLE
How to force refresh outline?

### DISCUSSION THREAD

**Discussion by shutaozhenzhen** [microsoft/vscode-cpptools#9299] [Q&A] [answered]
I am working with a project control in Git. This peoject has many branchs, and I need to work with most of them.
When I checkout to another branch, outline stay with the pre outline.

So, how to force refresh outline?

**Comment by sean-mcmanus** [comment]
The file watcher is supposed to detect the file changes and reparse the files and cause the outline to update. If not, you can try running the C/C++: Rescan Workspace command from the Command Palette. Does that work?

**Reply by sean-mcmanus** [reply]
Okay, I reproed the issue -- VS Code isn't sending us a provideDocumentSymbols request until you close/reopen the document. I don't know yet if there's a way for us to trigger that event ourselves.

**Reply by sean-mcmanus** [reply]
The issue is being tracked by VS Code: https://github.com/microsoft/vscode/issues/108722 -- you can upvote that issue.

**Reply by shutaozhenzhen** [reply]
Rescan Workspace command is useful. It just need one more step: close current file and reopen it, otherwise outline will show "can not find symbol"

**Comment by sean-mcmanus** [comment, marked as answer]
The issue is being tracked by VS Code: https://github.com/microsoft/vscode/issues/108722 -- you can upvote that issue.
